### PR TITLE
Throwables helper to help fatal checking of Throwable for Java

### PR DESCRIPTION
**Pull request by He-Pin**
…Java,alias with Scala's Nonfatal pattern matching.

### CODE REVIEW

File: akka-actor/src/main/scala/akka/util/Throwables.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.util
+
+import scala.util.control.NonFatal
+
+object Throwables {
```
Comment (ktoso) [inline akka-actor/src/main/scala/akka/util/Throwables.scala]: needs docs why this exists and what the intended use case is -- to allow catching only non fatal ones in supervision on routing in akka http I suppose

**Review by ktoso — APPROVED** [review]
I can see this adding value to our Java DSL offering... perhaps it should be in a javadsl package though, or `akka.japi`

**Comment by hepin1989** [comment]
Yes，I extracted it from our inner project, it's really needed when we want to achieve the same sematic as it is in Scala.

`japi` is a good fit place too.

**Comment by hepin1989** [comment]
@ktoso Ping

**Comment by ktoso** [comment]
No need to ping, already got 2 emails before you pinged. Things will be reviewed in their time.

**Comment by hepin1989** [comment]
Haha, Sorry for the bother:)

**Comment by johanandren** [comment]
I can see how it is useful, but does it really belong in Akka?

**Comment by hepin1989** [comment]
And then use it as 
image

We don't need this in Scala:)

**Comment by johanandren** [comment]
That does not change my assessment, I see that it is useful, but is it really something a toolkit for concurrency and distributed systems should provide, as I see it it is filling out a void in the standard library rather (and we can't provide fillers for all those gaps in the scope of Akka)

**Comment by patriknw** [comment]
What about using `scala.util.control.NonFatal` instead? The only difference is the naming:

```java
import scala.util.control.NonFatal;

   try {
     // dangerous stuff
   } catch(Throwable e) {
     if (NonFatal.apply(e)){
       log.error(e, "Something not that bad.");
     } else {
       throw e;
     }
```

**Comment by ktoso** [comment]
The NonFatal PR (this one) I can see being valuable, since Java people tend to panic once they have to touch a scala class. The other PR I think was going too far, but this I'd be ok with. Esp if wanting to implement same semantics things in Java, for Alpakka or Akka HTTP, this method would help.

Unless someone feels strongly against it

**Comment by patriknw** [comment]
I'm ok with adding it. Will comment on the implementation, though.

File: akka-actor/src/main/scala/akka/japi/Throwables.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ *       throw e;
+ *     }
+ * }}}
+ */
+object Throwables {
+  /**
+   * Returns true if the provided `Throwable` is to be considered non-fatal,
+   * or false if it is to be considered fatal
+   */
+  def isNonFatal(throwable: Throwable): Boolean = throwable match {
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/japi/Throwables.scala]: can be:
```
def isNonFatal(throwable: Throwable): Boolean = NonFatal(throwable)
```

**Review by patriknw — CHANGES_REQUESTED** [review]

File: akka-actor/src/main/scala/akka/japi/Throwables.scala (hunk 2 above)
Comment (ktoso) [reply akka-actor/src/main/scala/akka/japi/Throwables.scala]: Good one!

**Review by ktoso — APPROVED** [review]
Was amended, LGTM

**Pull request merged**